tests/phpunit/api/v3/ContributionTest.php

```diff
@@ -2326,6 +2328,43 @@ public function testRepeatTransaction() {
     ], 'online');
   }
 
+  /**
+   * Test custom data is copied over.
+   *
```

Comment:
Hmm - Ok  I did start from the assumption that what https://github.com/civicrm/civicrm-core/pull/17454 was trying to do was right & that I just needed to write a test & fix the implementation to get it over the line 

I could make a case that the API should require the template contribution to be passed in and just use that. However, we are already calculating the template contribution in the BAO function for line items so I think it makes sense we should do it for facets. I think reversing course doesn't make sense because at least some processors only 'know' the first contribution & are already passing that in

Reply:
I've updated the comment block

Reply:
@artfulrobot the contribution we are using is the one retrieved by the ```CRM_Contribute_BAO_ContributionRecur::getTemplateContribution``` function - which will retrieve an 'actual' template contribution if there is one, but which will get most recent otherwise.

(from Barcelona sprint)

https://github.com/civicrm/civicrm-core/pull/15456

Reply:
@artfulrobot In the context of this PR we should ignore `"which contribution should be used as a template for a repeat contribution belonging to a recurring contribution"`. That is an issue but it's covered by https://github.com/civicrm/civicrm-core/pull/17455